File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
    if(nums.length == 0) return 0
    let current = 0, n = nums.length;

    for(let i = 0; i < nums.length; i++) {
        if(nums[i] != nums[current]){
            current++
            nums[current] = nums[i]
        }
    }

    return current + 1
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;

        int k = 1, i = 0, j = 1;

        while (j < n) {
            if (nums[j] != nums[j - 1]) {
                k++;
                nums[i+1] = nums[j];
                i++;
                j++;
            } else {
                j++;
            }
        }

        return k;
    }
}